lib/PlacesProvider.js

```diff
@@ -273,6 +273,16 @@ Links.prototype = {
   },
 
   /**
+   * Removes a bookmark
+   *
+   * @param {String} bookmarkGuid the bookmark guid
+   * @returns {Promise} Returns a promise set to an object representing the removed bookmark
+   */
+  deleteBookmark: function PlacesProvider_deleteBookmark(bookmarkGuid) {
```

Comment:
nit: we're trying something from now on: every function that is async (i.e. returns a promise) should have `async` in the name.

this method would be `asyncDeleteBookmark`.

We haven't applied the same convention to other functions yet, and we should.